google/ggrc-core
# GGRC-7090 "Get Permalink" should appear under 3b menu in case filter was not saved on "Advanced search popup"

# Issue description

"Get Permalink" should also appear in case tree view was filtered with a query that was not saved on "Advanced search popup". Meaning each time when user searched something with advanced search he should see under three dots an option "Get Permalink". He can click it and link is copied at clipboard. Then it can be shared to another user.

# Steps to test the changes

1. Open "Advanced search popup".
2. Set filter options and click Apply button.
3. Verify that "Get Permalink" button is shown.
4. Click "Get Permalink" button.
5. Verify that save_search request was sent. And permalink was saved to clipboard. 

# Solution description

BE part of the solution includes following changes:

- `is_visible` column is added on `saved_searches table`. This flag helps to sort out saved searches that are created with "Get Permalink" but are not saved as a saved search explicitly.

- `GET /api/saved_searches/<string:search_type>` is modified so now it will return only saved searches that have `is_visible` set to `True`.

- `POST /api/saved_searches` is modified:

     - New argument `is_visible` is accepted;
     - `name` argument is not obligatory anymore if `is_visible: False` is passed. In this case name of saved search will be autogenerated.

FE part of the solution includes following changes:

- On click Apply button search filter options is saved in a instance.
- 'Get Permalink' button saves search with `is_visible` set to `false` on first click and provides permalink. 

# Sanity checklist




# PR Review checklist
